src/I18nExtension.php

```diff
@@ -5,42 +5,66 @@
  *
  * (c) 2010-2019 Fabien Potencier
  * (c) 2019 phpMyAdmin contributors
+ * (c) 2021 Christopher Marshall contributor
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
  */
 
-namespace PhpMyAdmin\Twig\Extensions;
+namespace PhpMyAdmin\Twig\Extensions
+{
+	use PhpMyAdmin\Twig\Extensions\TokenParser\TransTokenParser;
+	use Twig\Extension\AbstractExtension;
+	use Twig\TwigFilter;
 
-use PhpMyAdmin\Twig\Extensions\TokenParser\TransTokenParser;
-use Twig\Extension\AbstractExtension;
-use Twig\TwigFilter;
+	class I18nExtension extends AbstractExtension
+	{
+		/**
+		 * {@inheritdoc}
+		 */
+		public function getTokenParsers()
+		{
+			return [new TransTokenParser()];
+		}
 
-class I18nExtension extends AbstractExtension
-{
-    /**
-     * {@inheritdoc}
-     */
-    public function getTokenParsers()
-    {
-        return [new TransTokenParser()];
-    }
+		/**
+		 * {@inheritdoc}
+		 */
+		public function getFilters()
+		{
+			return [
+				new TwigFilter('trans', '\twig_gettext_translate'), /* Note, the filter does not handle plurals */
```

Comment:
I now understand !
That said I am "sure" you can pass a callback to TwigFilter and that would avoid adding a namespace function

Alternative: add the function to this class and make it static and then call it with I18nExtension::functionName

Just want to avoid to have this dangling namespace ;)

Reply:
You know, I actually originally did a method in the class before I later changed it to the Namespace thinking I was actually making things better :)

I've reverted it back to a static function in the class. IMO I just don't feel that decoupling the dependency is worth the gain because it's so marginal.